update hardware requirement for disaggregated tiflash (#16642)
pingcap/docs: hardware-and-software-requirements.md

## Web browser requirements

TiDB relies on [Grafana](https://grafana.com/) to provide visualization of database metrics. A recent version of Microsoft Edge, Safari, Chrome or Firefox with Javascript enabled is sufficient.

## Hardware and software requirements for TiFlash disaggregated storage and compute architecture

The preceding TiFlash software and hardware requirements are for the coupled storage and compute architecture. Starting from v7.0.0, TiFlash supports the [disaggregated storage and compute architecture](/tiflash/tiflash-disaggregated-and-s3.md). In this architecture, TiFlash is divided into two types of nodes: the Write Node and the Compute Node. The requirements for these nodes are as follows:

- Software: remain the same as the coupled storage and compute architecture, see [OS and platform requirements](#os-and-platform-requirements).
- Network port: remain the same as the coupled storage and compute architecture, see [Network](#network-requirements).
- Disk space:

>     - TiFlash Write Node: it is recommended to configure 200 GB+ of disk space, which is used as a local buffer when adding TiFlash replicas and migrating Region replicas before uploading data to Amazon S3. In addition, an object storage compatible with Amazon S3 is required.
>     - TiFlash Compute Node: it is recommended to configure 100 GB+ of disk space, which is mainly used to cache the data read from the Write Node to improve performance. The cache of the compute node may be fully used, which is normal.

Reviewer: ```suggestion
    - TiFlash Write Node: it is recommended to configure at least 200 GB of disk space, which is used as a local buffer when adding TiFlash replicas and migrating Region replicas before uploading data to Amazon S3. In addition, an object storage compatible with Amazon S3 is required.
    - TiFlash Compute Node: it is recommended to configure at least 100 GB of disk space, which is mainly used to cache the data read from the Write Node to improve performance. The cache of the Compute Node might be fully used, which is normal.
```

- CPU and memory requirements are as follows.

### Development and test environments

| Component | CPU | Memory | Local Storage | Network | Number of Instances (Minimum Requirement) |
| --- | --- | --- | --- | --- | --- |

> | TiFlash Write Node | 16 cores+ | 32 GB+ | SSD, 200 GB+ | Gigabit Ethernet | 1 |
> | TiFlash Compute Node | 16 cores+ | 32 GB+ | SSD, 100 GB+ | Gigabit Ethernet | 0 (see the following note) |

Reviewer: ```suggestion
| TiFlash Write Node | 16 cores+ | At least 32 GB | SSD, 200 GB+ | Gigabit Ethernet | 1 |
| TiFlash Compute Node | 16 cores+ | At least 32 GB | SSD, 100 GB+ | Gigabit Ethernet | 0 (see the following note) |
```

### Production environment

| Component | CPU | Memory | Disk Type | Network | Number of Instances (Minimum Requirement) |
| --- | --- | --- | --- | --- | --- |

> | TiFlash Write Node | 32 cores+ | 64 GB+ | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 1 |
> | TiFlash Compute Node | 32 cores+ | 64 GB+ | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 0 (see the following note) |

Reviewer: ```suggestion
| TiFlash Write Node | At least 32 cores | At least 64 GB | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 1 |
| TiFlash Compute Node | At least 32 cores | At least 64 GB | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 0 (see the following note) |
```

4. CPU and Other Memory Requirements - Refer to the Next Section

### Development and Testing Environment

| **Component** | **CPU** | **Memory** | **Local Storage** | **Network** | **Instance Quantity (Minimum Requirement)** |
| --- | --- | --- | --- | --- | --- |
| TiFlash Storage Node | 16 cores+ | 32 GB+ | SSD, 200 GB+ | Gigabit Ethernet | 1 |
| TiFlash Compute Node | 16 cores+ | 32 GB+ | SSD, 100 GB+ | Gigabit Ethernet | 0 |

### Production Environment

| **Component** | **CPU** | **Memory** | **Disk Type** | **Network** | **Instance Quantity (Minimum Requirement)** |
| --- | --- | --- | --- | --- | --- |
| TiFlash Storage Node | 32 cores+ | 64 GB+ | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 1 |
| TiFlash Compute Node | 32 cores+ | 64 GB+ | 1 or more SSDs | 10 Gigabit Ethernet (2 recommended) | 0 |

> **Note:**
>
> - TiFlash Compute Nodes can be quickly scaled in or out using deployment tools such as TiUP, with a scaling range of [0, inf+].

> 

Reviewer 2: ```suggestion
```
